File: daily_problems/2024/07/0724/personal_submission/cf798c_fancyalad.py
# submission link: https://codeforces.com/problemset/submission/798/272248345
# gcd(x-y,x+y)=gcd(x-y,(x+y)-(x-y))=gcd(x-y,2y)
# gcd(x-y,x+y)=gcd(x-y,(x+y)+(x-y))=gcd(x-y,2x)
# (x, y)->(x-y, x+y)->((x+y)-(x-y), (x+y)+(x-y))=(2x, 2y)

import sys
import math
input = lambda: sys.stdin.readline().strip()

n = int(input())
nums = list(map(int, input().split()))

print("YES")
if math.gcd(*nums) > 1:
    print(0)
else:
    ans = cnt1 = 0
    for x in nums:
        if x % 2:
            cnt1 += 1
        else:
            # 如果是一奇一偶相邻，比如 3 和 4，进行一次操作变成 1，7，第二次变成 6，8，在又变成 2，14
            # 如果两个奇数，比如 3 和 5，第一次 2，8，第二次 6，10
            ans += cnt1 // 2 + cnt1 % 2 * 2
            cnt1 = 0
    ans += cnt1 // 2 + cnt1 % 2 * 2
    print(ans)

File: daily_problems/2024/07/0724/personal_submission/cf798c_fancyalad.cpp
// submission link: https://codeforces.com/contest/798/submission/272251131

#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n, g = 0;
    cin >> n;
    vector<int> nums (n);
    int cnt = 0, ans = 0;
    for (auto& num : nums) {
        cin >> num;
        g = __gcd(g, num);
        if (num % 2) {
            cnt ++;
        } else {
            ans += cnt / 2 + cnt % 2 * 2;
            cnt = 0;
        }
    }
    cout << "YES" << endl;
    if (g > 1) {
        cout << 0 << endl;
    } else {
        ans += cnt / 2 + cnt % 2 * 2;
        cout << ans << endl;
    }
    return 0;
}